x/erc20/simulation/proposal.go

```diff
@@ -0,0 +1,259 @@
+// Simulation operation weights constants
+const (
+	DefaultWeightMsgUpdateParams int = 100
+
+	OpWeightMsgUpdateParams                       = "op_weight_msg_update_params"
+	OpWeightSimulateRegisterCoinProposal          = "op_weight_register_coin_proposal"
+	OpWeightSimulateRegisterERC20Proposal         = "op_weight_register_erc20_proposal"
+	OpWeightSimulateToggleTokenConversionProposal = "op_weight_toggle_token_conversion_proposal"
+
+	erc20Decimals = uint8(18)
+)
+
+// ProposalMsgs defines the module weighted proposals' contents
+func ProposalMsgs(k keeper.Keeper, ak types.AccountKeeper, bk types.BankKeeper, ek types.EVMKeeper, fk types.FeeMarketKeeper) []simtypes.WeightedProposalMsg {
+	return []simtypes.WeightedProposalMsg{
+		simulation.NewWeightedProposalMsg(
+			OpWeightMsgUpdateParams,
+			DefaultWeightMsgUpdateParams,
+			SimulateMsgUpdateParams,
+		),
+		simulation.NewWeightedProposalMsg(
+			OpWeightSimulateRegisterCoinProposal,
+			params.DefaultWeightRegisterCoinProposal,
+			SimulateMsgRegisterCoin(k, bk),
+		),
+		simulation.NewWeightedProposalMsg(
+			OpWeightSimulateRegisterERC20Proposal,
+			params.DefaultWeightRegisterERC20Proposal,
+			SimulateMsgRegisterERC20(k, ak, bk, ek, fk),
+		),
+		simulation.NewWeightedProposalMsg(
+			OpWeightSimulateToggleTokenConversionProposal,
+			params.DefaultWeightToggleTokenConversionProposal,
+			SimulateMsgToggleTokenConversion(k, bk, ek, fk),
+		),
+	}
+}
+
+// SimulateMsgUpdateParams returns a random MsgUpdateParams
+func SimulateMsgUpdateParams(r *rand.Rand, _ sdk.Context, _ []simtypes.Account) sdk.Msg {
+	// use the default gov module account address as authority
+	var authority sdk.AccAddress = address.Module("gov")
+
+	params := types.DefaultParams()
+
+	params.EnableErc20 = generateRandomBool(r)
+	params.EnableEVMHook = generateRandomBool(r)
+
+	return &types.MsgUpdateParams{
+		Authority: authority.String(),
+		Params:    params,
+	}
+}
+
+func SimulateRegisterCoin(r *rand.Rand, ctx sdk.Context, accs []simtypes.Account, k keeper.Keeper, bk types.BankKeeper) (sdk.Msg, error) {
+	coinMetadata := types.GenRandomCoinMetadata(r)
+	if err := bk.MintCoins(ctx, types.ModuleName, sdk.NewCoins(sdk.NewCoin(coinMetadata.Base, sdkmath.NewInt(1000000000000)))); err != nil {
+		panic(err)
+	}
+	bankparams := bk.GetParams(ctx)
+	bankparams.DefaultSendEnabled = true
+	bk.SetParams(ctx, bankparams)
+
+	params := k.GetParams(ctx)
+	params.EnableErc20 = true
+	k.SetParams(ctx, params)
+
+	// mint cosmos coin to random accounts
+	randomIteration := r.Intn(10)
+	for i := 0; i < randomIteration; i++ {
+		simAccount, _ := simtypes.RandomAcc(r, accs)
+
+		if err := bk.MintCoins(ctx, types.ModuleName, sdk.NewCoins(sdk.NewCoin(coinMetadata.Base, sdkmath.NewInt(1000000000)))); err != nil {
+			return &types.MsgRegisterCoin{}, err
+		}
+		if err := bk.SendCoinsFromModuleToAccount(ctx, types.ModuleName, simAccount.Address, sdk.NewCoins(sdk.NewCoin(coinMetadata.Base, sdkmath.NewInt(1000000000)))); err != nil {
```

Comment:
@poorphd 
Since `sdkmath.NewInt(1000000000)` is a commonly used value, how about defining it as a variable `amt` outside the for loop? You could then use `amt` within the loop.

Reply:
applied in https://github.com/b-harvest/Canto/commit/542ac822dc52a0bd9d69bea1583696fc3fcc4d5c